presto-main/src/main/java/io/prestosql/execution/QueryTracker.java

```diff
@@ -46,10 +54,12 @@
 import static java.util.Objects.requireNonNull;
 
 @ThreadSafe
-public class QueryTracker<T extends TrackedQuery>
+public class QueryTracker<T extends QueryExecution>
```

Comment:
Today `QueryTracker` is only instantiated as `QueryTracker<QueryExecution` so this change is ok.
But to make this future-rebase-proof maybe don't impact this signature and use something like:
```
    private Optional<QueryExecution> tryGetQueryExecution(QueryId queryId)
    {
        return tryGetQuery(queryId)
                .filter(QueryExecution.class::isInstance)
                .map(QueryExecution.class::cast);
    }
```